### DISCUSSION TITLE
OAuth out-of-band (oob) flow deprecated 28 Feb 2022

### DISCUSSION THREAD

**Discussion by ialexs** [nithinmurali/pygsheets#530] [Q&A]
Hello all.. 

From https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#instructions-oob

> New clients will be unable to use this flow starting on Feb 28, 2022 with the `redirect_uri=urn:ietf:wg:oauth:2.0:oob` or `urn:ietf:wg:oauth:2.0:oob:auto or oob`

Even I've define the `redirect_uris` in the `client_secret.json` to `http://localhost/` , 
pygsheets still give auth URL with with `redirect_uri` to `urn:ietf:wg:oauth:2.0:oob:auto or oob`

So I change `redirect_uri='urn:ietf:wg:oauth:2.0:oob'` to `redirect_uri='http://localhost/'` in the `authorization.py`

It works again... :) Am I doing this right? :D

**Comment by nithinmurali** [comment]
Thank you for notifying. i will fix the flow.

**Reply by francot** [reply]
I open issue https://github.com/nithinmurali/pygsheets/issues/586

**Reply by rrajan** [reply]
@nithinmurali can you please summarize the fix for this to a layman. what all needs to be done to get this working? I also see stuff that google now wants links to privacy policy and homepage on the oauth consent page - would you know what I need for this?
my use case is just a simple python code that reads and writes to gsheets
